plugins/user_mapping/src/interface.h

```diff
@@ -0,0 +1,53 @@
+#ifndef IRODS_HTTP_API_USER_MAPPER_INTERFACE_H
+#define IRODS_HTTP_API_USER_MAPPER_INTERFACE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Initializes the user mapping plugin.
+///
+/// \param[in] _args A C-string containing the JSON representing the configuration for the plugin.
+///
+/// \pre \p _args must be a non-null C-string that represents a JSON structure.
+///
+/// \returns A code representing the status of the plugin.
+/// \retval  zero if initialization of the plugin was succsessful.
+/// \retval  non-zero if initialization of the plugin was not succsessful.
+int user_mapper_init(const char* _args);
+
+/// Matches the given information to a user.
+///
+/// \param[in]  _param A C-string containing the JSON representing information from an
+///                    authenticated OpenID User.
```

Comment:
Same thing here: Would it be worth documenting said JSON structure here as well?

Reply:
Documenting a specific example, I'm not too sure about, since it depends on the OpenID Provider's configuration. We could be more explicit and state that the function accepts either an OpenID access token or an OpenID ID token.

Reply:
I misunderstood what was happening here. Maybe just mention that the form of the JSON depends on what is provided by the OpenID provider and that should be enough. Thoughts?

Reply:
I think we can add `This can either take the form of an OpenID Access Token or ID Token.` or something similar to the end.

Reply:
Used the given line.